### PR TITLE
Updates from onnx-pytorch: Python 3 compatibility, and some extra test flexibility

### PR DESCRIPTION
**Pull request by ezyang**

### CODE REVIEW

File: test/common.py [hunk 1]
```diff
@@ -283,7 +283,7 @@ def remove_prefix(text, prefix):
                 return text[len(prefix):]
             return text
         munged_id = remove_prefix(self.id(), "__main__.")
-        expected_file = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+        expected_file = os.path.join(os.path.dirname(os.path.realpath(__main__.__file__)),
```
Comment (apaszke) [inline test/common.py]: Why is this needed?

File: torch/nn/_functions/rnn.py [hunk 2]
```diff
@@ -74,7 +81,11 @@ def forward(input, hidden, weight):
         next_hidden = []
 
         if lstm:
-            hidden = list(zip(*hidden))
+            # NB: Be sure to use the generator version of zip here in
+            # Python 2, since the evaluation order differs (Python 2
+            # evaluates all of the first argument, and then evaluates
+            # all of the second argument; generator version interleaves.)
```
Comment (apaszke) [inline torch/nn/_functions/rnn.py]: Uh how does the argument evaluation order matter here?

**Review by prigoyal — APPROVED** [review]

File: test/common.py (hunk 1 above)
Comment (ezyang) [reply test/common.py]: Without it, we always place expect files in the same place as `test/common.py` We don't want this for onnx-pytorch tests, which don't live in the PyTorch test directory.

File: torch/nn/_functions/rnn.py (hunk 2 above)
Comment (ezyang) [reply torch/nn/_functions/rnn.py]: Because the index here is a *tensor indexing* expression, which is an autograd operation which shows up in the trace.

Comment (apaszke) [reply torch/nn/_functions/rnn.py]: Uh it sounds more like sth that should be fixed in tracer tests. Both are valid traces, so there's no need to enforce full determinism in all library code.

**Comment by ezyang** [comment]
This commit set doesn't fix the root problem; closing.
